data/lib/src/datasource/authentication_datasource.dart

```diff
@@ -45,18 +45,22 @@ class AuthenticationDataSource {
 
   AuthenticationDataSource(this.linShareHttpClient, this.deviceManager, this._remoteExceptionThrower);
 
-  Future<Token> createPermanentToken(Uri baseUrl, UserName userName, Password password) async {
+  Future<Token> createPermanentToken(Uri baseUrl, UserName userName, Password password, {OTPCode otpCode}) async {
     return Future.sync(() async {
       final deviceUUID = await deviceManager.getDeviceUUID();
       final permanentToken = await linShareHttpClient.createPermanentToken(
           baseUrl,
           userName.userName,
           password.value,
-          PermanentTokenBodyRequest('LinShare-${deviceManager.getPlatformString()}-$deviceUUID'));
+          PermanentTokenBodyRequest('LinShare-${deviceManager.getPlatformString()}-$deviceUUID'),
+          otpCode: otpCode);
       return permanentToken.toToken();
     }).catchError((error) {
         _remoteExceptionThrower.throwRemoteException(error, handler: (DioError error) {
           if (error.response.statusCode == 401) {
+            if (isAuthenticateWithOTPError(error.response.headers.value('X-Linshare-Auth-Error-Code'))) {
```

Comment:
`X-Linshare-Auth-Error-Code` why not constant it?

Reply:
done